hal/src/sercom/v2/i2c.rs

```diff
@@ -0,0 +1,137 @@
+//! Use the SERCOM peripheral for I2C communications.
+
+mod pads;
+pub use pads::*;
+
+mod reg;
+use reg::Registers;
+
+mod flags;
+pub use flags::*;
+
+mod config;
+pub use config::*;
+
+pub mod impl_ehal;
+
+/// Size of the SERCOM's `DATA` register
+#[cfg(any(feature = "samd11", feature = "samd21"))]
+pub type DataReg = u16;
+
+/// Size of the SERCOM's `DATA` register
+#[cfg(any(feature = "min-samd51g"))]
+pub type DataReg = u32;
+
+/// Inactive timeout configuration
+#[repr(u8)]
+#[derive(Clone, Copy)]
+pub enum InactiveTimeout {
+    /// Disabled
+    Disabled = 0x0,
+    /// 5-6 SCL cycles (50-60 us @ 100 kHz)
+    Us55 = 0x1,
+    ///10-11 SCL cycles (100-110 us @ 100 kHz)
+    Us105 = 0x2,
+    /// 20-21 SCL cycles (200-210 us @ 100 kHz)
+    Us205 = 0x3,
+}
+
+pub struct I2c<C: AnyConfig> {
+    config: C,
+}
+
+impl<C: AnyConfig> I2c<C> {
```

Comment:
Do you want some way to `unsafe`ly write the DATA register directly? It also might help to expose a way to get a pointer to the DATA register, for setting up DMA with `samd-dma` rather than the `dmac` module.